File: Level-0/56.java
class Solution {
    public String solution(String polynomial) {
        String answer = "";
        String[] s = polynomial.trim().split(" \\+ ");
        int[] num = new int[2];
        for(int i = 0; i < s.length; i++){
            if(s[i].contains("x")){
                if(s[i].length() == 1) s[i] = s[i].replace("x", "1");
                else s[i] = s[i].replace("x", "");
                num[0] += Integer.parseInt(s[i]);
            }else{
                num[1] += Integer.parseInt(s[i]);
            }
        }
        
        if(num[1] == 0){
            if(num[0] == 1) answer = "x";
            else answer = String.valueOf(num[0]) + "x";
        }else if(num[0] == 0){
            answer = String.valueOf(num[1]);
        }else{
            if(num[0] == 1) answer = "x + " + String.valueOf(num[1]);
            else answer = String.valueOf(num[0]) + "x + " + String.valueOf(num[1]);
        }
        
        return answer;
    }
}

File: Level-0/56.js
function solution(polynomial) {
    const arr = polynomial.split(" + ");
    const xNum = arr.filter(n => n.includes("x"))
                    .map(n => Number(n.replace("x", "")) || 1)
                    .reduce((acc, cur) => acc + cur, 0);
    const num = arr.filter(n => !isNaN(n))
                   .reduce((acc, cur) => acc + parseInt(cur, 10), 0);
    
    let answer = [];
    if(xNum) answer.push(`${xNum === 1 ? "" : xNum}x`);
    if(num) answer.push(num);
    
    return answer.join(" + ");
}
